### PR TITLE
Fetch subsequent paginated data on `list_*` functions

### PR DESCRIPTION
**Pull request by ghost**
This PR addresses #80.

### CODE REVIEW

File: integration_tests/test_api.py [hunk 1]
```diff
@@ -347,14 +347,21 @@ async def test_list_gridpool_trades(set_up: dict[str, Any]) -> None:
 
 
 @pytest.mark.asyncio
-async def test_list_public_trades(set_up: dict[str, Any]) -> None:
+async def test_list_public_trades_pages(set_up: dict[str, Any]) -> None:
     """Test listing public trades."""
-    public_trades = await set_up["client"].list_public_trades(
+    public_trades = []
+    page_limit = 3  # Limit the test to 3 pages
```
Comment (Marenz) [inline integration_tests/test_api.py]: If we only assert for len > 0, then we don't need to wait for three pages in, no?

Comment (Marenz) [reply integration_tests/test_api.py]: Actually, we even allow len >= 0, what exactly are we testing here :)

Comment (ghost) [reply integration_tests/test_api.py]: Yea good call this was an old check but now I could just use a delivery period for which I know that there are enough public trades and ensure that I get the expected amount of elements back. Updated now

**Comment by ghost** [comment]
I now implemented it for the other `list_*` functions too

**Review by cwasicki — APPROVED** [review]

File: src/frequenz/client/electricity_trading/_client.py [hunk 2]
```diff
@@ -799,58 +799,57 @@ async def list_gridpool_orders(
...
-                try:
-                    orders.append(OrderDetail.from_pb(order_detail))
-                except InvalidOperation:
-                    _logger.error(
-                        "Failed to convert order details for order: %s",
-                        str(order_detail).replace("\n", ""),
+                for order_detail in response.order_details:
+                    yield OrderDetail.from_pb(order_detail)
+
+                if len(response.pagination_info.next_page_token):
```
Comment (matthias-wende-frequenz) [inline src/frequenz/client/electricity_trading/_client.py]: You don't need the `len` here. An empty string is false.

Comment (ghost) [reply src/frequenz/client/electricity_trading/_client.py]: done

**Review by matthias-wende-frequenz — APPROVED** [review]

**Pull request merged**